### PR TITLE
Added hasBestPrice handler

### PR DESCRIPTION
**Pull request by EduardoRodriguesF**
This new condition option allows developers to create conditions whether the product is on promotion.

#### What problem is this solving?

If a developer wanted to create a different layout for products on promotion, they would need to create a custom component to do that.

This Pull Request implements this solution on `condition-layout.product` block by adding a new condition handler.

#### How to test it?

Create a condition layout like the following:

```json
{
  "condition-layout.product": {
    "props": {
      "conditions": [
        {
          "subject": "hasBestPrice",
          "arguments": {
            "value": true
          }
        }
      ],
      "Then": "rich-text#discount",
      "Else": "rich-text#default"
    }
  },
  "rich-text#discount": {
    "props": {
      "text": "This product contains a discount!!"
    }
  },
  "rich-text#default": {
    "props": {
      "text": "Regular price."
    }
  },
}
```

Also, it is possible to use the shorthand version where no arguments is passed.

```json
{
  "condition-layout.product": {
    "props": {
      "conditions": [
        {
          "subject": "hasBestPrice"
        }
      ],
      "Then": "rich-text#discount",
      "Else": "rich-text#default"
    }
  }
}
```

Upon implementing it, you'll verify that it conditionally renders one over the other based on the list price and selling price diverging.

### CODE REVIEW

**Comment by mihainutiu-vtex** [comment]
Hi @EduardoRodriguesF 
Please resolve conflicts and I will publish a beta version to unblock you until this PR is approved and it goes to stable.

**Comment by mihainutiu-vtex** [comment]
Hey @EduardoRodriguesF, thanks for rebasing. I published a beta version for you to install on your account. 
`vtex.condition-layout@2.7.1-beta.0 `

Hope this helps!

**Comment by EduardoRodriguesF** [comment]
I just realized that my PR has the same intention as this other one: #40

How shall we proceed, @mihainutiu-vtex? Since my PR is more up-to-date with master I figure it makes sense to merge any missing cutting-edges that my implementation didn't cover. Would love to listen your thoughts though!

**Comment by mihainutiu-vtex** [comment]
> I just realized that my PR has the same intention as this other one: #40
> 
> How shall we proceed, @mihainutiu-vtex? Since my PR is more up-to-date with master I figure it makes sense to merge any missing cutting-edges that my implementation didn't cover. Would love to listen your thoughts though!

Hey @EduardoRodriguesF 
Indeed, you should cover any cutting-edges you may have missed. That older PR wasn't looked at in 2 years, so no point in waiting for it.

**Comment by EduardoRodriguesF** [comment]
@mihainutiu-vtex Done! I think I covered any missing aspects now.

**Review by mihainutiu-vtex — APPROVED** [review]
LGTM

**Pull request merged**
